src/core/fetchers/segment/segment_fetcher.ts

```diff
@@ -249,15 +251,15 @@ export default function createSegmentFetcher<TLoadedFormat, TSegmentDataType>(
 
       /**
        * Call a segment loader for the given URL with the right arguments.
-       * @param {string|null} url
+       * @param {Object|null} cdnMetadata
        * @param {Object} cancellationSignal
        * @returns {Promise}
        */
       function callLoaderWithUrl(
-        url : string | null,
+        cdnMetadata : ICdnMetadata | null,
```

Comment:
I'm not really sure about the usage of CDN as it's not the only provider we could use, I'd reckon using `mediaProvider` or something like that

Reply:
Yes I wasn't too sure about the name "cdn" but I found it was the one conveying the most the concept of a named entity delivering content through requests, 99% of them being referred as CDNs in most situations.
Thus I thought here that I would use this term as a sort of antonomasia to refer to all entities able to deliver content through requests with an URL.

`MediaProvider` is also good also but I'm still a little afraid that what we gain, we might lose on understandability as the name is a little more generic. Not only that, but I would say that it might be to broad, as this term could also be thought to refer to things like the loading logic of locally-stored contents - which does not rely on the `ICdnMetadata` interface as all this logic is only used for defining the URL to request (the loading logic of locally-stored contents do not use URLs but callbacks). When no URL is involved, it is `null` that is given here, which isn't a `ICdnMetadata`.
Though it's true that we could define a differently-structured `IMediaProvider` interface, also encompassing those other cases.

Also, I think that content steering is expected to also be used for things like Manifest requests, they might or might not be considered "media" here. I guess we could use also `ContentProvider` or even `ContentDistributor` but it still has some generic-ness and broadness I'm not too comfortable with. Still we could refactor other things so it appears more adapted.

What's your opinion on this?

Reply:
I totally agree.
Also, 👍 for using the term _antonomasia_

Reply:
I actually agree